### ISSUE TITLE
Wakaama can not compile with C90 compiler

### ISSUE THREAD

**Issue by fdur** [eclipse/wakaama#232]
Wakaama can not compile using a C90 compiler.
Most of the issues are linked to variable declaration in the middle of source code: intermingled declarations and code

Example:
packet.c
```                    // parse block1 header
                    uint32_t block1_num;
                    uint8_t  block1_more;
                    uint16_t block1_size;
                    coap_get_header_block1(message, &block1_num, &block1_more, &block1_size, NULL);
                    LOG_ARG("Blockwise: block1 request NUM %u (SZX %u/ SZX Max%u) MORE %u", block1_num, block1_size, REST_MAX_CHUNK_SIZE, block1_more);

                    // handle block 1
                    uint8_t * complete_buffer = NULL;
                    size_t complete_buffer_size;
```

For C90:
```                    uint32_t block1_num;
                    uint8_t  block1_more;
                    uint16_t block1_size;
                    uint8_t * complete_buffer = NULL;
                    size_t complete_buffer_size;

                    // parse block1 header
                    coap_get_header_block1(message, &block1_num, &block1_more, &block1_size, NULL);
                    LOG_ARG("Blockwise: block1 request NUM %u (SZX %u/ SZX Max%u) MORE %u", block1_num, block1_size, REST_MAX_CHUNK_SIZE, block1_more);

                    // handle block 1
                    complete_buffer = NULL;
```

**Comment by dnav** [comment]
Hi,

I should enforce separated declarations and code, at least for the sake of readability. But I can't see the benefits of being C90-compliant. C90 is 27-years old. Do you really need wakaama to be able to compile with this standard ?
For instance C90 does not allow C++ comments and wakaama is full of them.

Regards,

**Comment by jvermillard** [comment]
For some target the manufacturer provides only provides C90.
We have an example with an Intel target ;)

**Comment by jvermillard** [comment]
but our C90 compiler support C++ style comments

**Comment by dnav** [comment]
Please make a pull request with the required changes. I was afraid you wanted strict C90 compliance.

**Comment by jvermillard** [comment]
:+1:

**Comment by dnav** [comment]
Fixed by #252
